scripts/quests/outlands/An_Undying_Pledge.lua

```diff
@@ -0,0 +1,125 @@
+-----------------------------------
+-- An Undying Pledge
+-----------------------------------
+-- Log ID: 5, Quest ID: 149
+-- Stray Cloud : !pos 49 -6 15 252
+-- qm5         : !gotoid 17498653
+-----------------------------------
+require('scripts/globals/items')
+require('scripts/globals/keyitems')
+require('scripts/globals/npc_util')
+require('scripts/globals/quests')
+require('scripts/globals/zone')
+require('scripts/globals/interaction/quest')
+-----------------------------------
+local ID = require("scripts/zones/Sea_Serpent_Grotto/IDs")
+
+local quest = Quest:new(xi.quest.log_id.OUTLANDS, xi.quest.id.outlands.AN_UNDYING_PLEDGE)
+
+quest.reward =
+{
+    item = xi.items.LIGHT_BUCKLER,
+    fameArea = xi.quest.fame_area.NORG,
+    fame = 50,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and
+                player:getFameLevel(xi.quest.fame_area.NORG) >= 4
+        end,
+
+        [xi.zone.NORG] =
+        {
+            ['Stray_Cloud'] = quest:progressEvent(225),
+
+            onEventFinish =
+            {
+                [225] = function(player, csid, option, npc)
+                        quest:begin(player)
+                        quest:setVar(player, 'Prog', 1)
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.NORG] =
+        {
+            ['Stray_Cloud'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 1 then
+                        return quest:event(228)
+                    elseif player:hasKeyItem(xi.ki.CALIGINOUS_BLADE) then
+                        return quest:progressEvent(227)
+                    elseif quest:getVar(player, 'Prog') == 2 then
+                        return quest:event(229)
+                    else
+                        return player:startEvent(231)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [227] = function(player, csid, option, npc)
+                    if quest:complete(player) then
+                        player:delKeyItem(xi.ki.CALIGINOUS_BLADE)
+                    end
+                end,
+            },
+        },
+
+        [xi.zone.SEA_SERPENT_GROTTO] =
+        {
+            ['qm5'] =
+            {
+                onTrigger = function(player, npc)
+                    if
+                        quest:getVar(player, 'Prog') == 2 and
+                        player:getCharVar("anUndyingPledgeNM_killed") == 0 and
```

Comment:
this "anUndyingPledgeNM_killed" should be handled in the quest lua. See One_Good_Deed.lua for an example on how to call mob events in quest.lua

Moreover try and use quest vars only. Stuff like Stage, Prog, Option